content/guides/use-case/jupyter/run-jupyter.md

```diff
@@ -0,0 +1,175 @@
+
+### Run a JupterLab container with a bind mount
+
+Open a terminal, and change directory to a directory where you would like to save the notebook's data. Then, run the following command based on your operating system and shell.
+
+{{< tabs >}}
+{{< tab name="Windows (Command Prompt)" >}}
+
+```console
+$ docker run --rm -p 8888:8888 -v "%cd%":/home/jovyan/work jupyter/base-notebook start-notebook.py --NotebookApp.token='my-token'
+```
+{{< /tab >}}
+{{< tab name="Windows (PowerShell)" >}}
+
+```console
+$ docker run --rm -p 8888:8888 -v "$(pwd):/home/jovyan/work" jupyter/base-notebook start-notebook.py --NotebookApp.token='my-token'
+```
+
+{{< /tab >}}
+{{< tab name="Windows (Git Bash)" >}}
+
+```console
+$ docker run --rm -p 8888:8888 -v "/$(pwd):/home/jovyan/work" jupyter/base-notebook start-notebook.py --NotebookApp.token='my-token'
+```
+
+{{< /tab >}}
+{{< tab name="Mac / Linux" >}}
+
+```console
+$ docker run --rm -p 8888:8888 -v "$(pwd):/home/jovyan/work" jupyter/base-notebook start-notebook.py --NotebookApp.token='my-token'
+```
+
+{{< /tab >}}
+{{< /tabs >}}
+
+The `-v` option tells Docker to provide access to your machine's present working directory (PWD) from the container at `/home/jovyan/work`.
+
+To access the container, in a web browser navigate to
+[localhost:8888/lab?token=my-token](http://localhost:8888/lab?token=my-token).
+Notebooks will now be saved to your local computer and will accessible even when
+the container is deleted.
+
+### Create and save a notebook
+
+For this example, you'll use the [Iris Dataset](https://scikit-learn.org/stable/auto_examples/datasets/plot_iris_dataset.html) example from sickit-learn.
+
+1. In the JupterLab interface, double click on the **work** folder in the left
+   sidebar. This lets you save your work to the bind mount you created.
+
+2. In the **Launcher**, under **Notebook**, select **Python 3**.
+
+3. In the notebook, specify the following to install the necessary package.
+
+   ```console
+   !pip install matplotlib scikit-learn
+   ```
+
+4. Select the play button to run the code.
+
+5. In the notebook, specify the following code.
+   ```python
+   from sklearn import datasets
+
+   iris = datasets.load_iris()
+   import matplotlib.pyplot as plt
+
+   _, ax = plt.subplots()
+   scatter = ax.scatter(iris.data[:, 0], iris.data[:, 1], c=iris.target)
+   ax.set(xlabel=iris.feature_names[0], ylabel=iris.feature_names[1])
+   _ = ax.legend(
+      scatter.legend_elements()[0], iris.target_names, loc="lower right", title="Classes"
+   )
+   ```
+6. Select the play button to run the code. You should see a scatter plot of the
+   Iris dataset.
+
+7. In the top menu, select **File** and then **Save Notebook As...**.
+
+8. Specify a name in the `work` directory. For example, `work/mynotebook.ipynb`.
+
+9. Select **Save**.
+
+The notebook is now saved on your host machine in the directory that you bind
+mounted.
+
+In the terminal, press `ctrl`+ `c` to stop the container.
```

Comment:
At this point, my terminal had a bunch of errors. Nothing seemed broken, the notebook was saved and it was there when I restarted the container.

```
[E 2024-02-22 09:08:54.285 ServerApp] Uncaught exception POST /api/contents/work/Untitled.ipynb/checkpoints?1708592934278 (192.168.65.1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='POST', uri='/api/contents/work/Untitled.ipynb/checkpoints?1708592934278', version='HTTP/1.1', remote_ip='192.168.65.1')
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.11/site-packages/tornado/web.py", line 1786, in _execute
        result = await result
                 ^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/handlers.py", line 304, in post
        checkpoint = await ensure_async(cm.create_checkpoint(path))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 189, in ensure_async
        result = await obj
                 ^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/manager.py", line 1066, in create_checkpoint
        return await self.checkpoints.create_checkpoint(self, path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/filecheckpoints.py", line 140, in create_checkpoint
        return await self.checkpoint_model(checkpoint_id, dest_path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/filecheckpoints.py", line 150, in checkpoint_model
        stats = await run_sync(os.stat, os_path)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/to_thread.py", line 33, in run_sync
        return await get_async_backend().run_sync_in_worker_thread(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2106, in run_sync_in_worker_thread
        return await future
               ^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 833, in run
        result = context.run(func, *args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
    FileNotFoundError: [Errno 2] No such file or directory: '/home/jovyan/work/.ipynb_checkpoints/Untitled-checkpoint.ipynb'
[W 2024-02-22 09:08:54.286 ServerApp] wrote error: 'Unhandled error'
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.11/site-packages/tornado/web.py", line 1786, in _execute
        result = await result
                 ^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/handlers.py", line 304, in post
        checkpoint = await ensure_async(cm.create_checkpoint(path))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 189, in ensure_async
        result = await obj
                 ^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/manager.py", line 1066, in create_checkpoint
        return await self.checkpoints.create_checkpoint(self, path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/filecheckpoints.py", line 140, in create_checkpoint
        return await self.checkpoint_model(checkpoint_id, dest_path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/contents/filecheckpoints.py", line 150, in checkpoint_model
        stats = await run_sync(os.stat, os_path)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/to_thread.py", line 33, in run_sync
        return await get_async_backend().run_sync_in_worker_thread(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2106, in run_sync_in_worker_thread
        return await future
               ^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 833, in run
        result = context.run(func, *args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
    FileNotFoundError: [Errno 2] No such file or directory: '/home/jovyan/work/.ipynb_checkpoints/Untitled-checkpoint.ipynb'
[E 2024-02-22 09:08:54.287 ServerApp] {
      "Host": "localhost:8888",
      "Accept": "*/*",
      "Referer": "http://localhost:8888/lab/workspaces/auto-H/tree/work/Untitled.ipynb",
      "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36"
    }
[E 2024-02-22 09:08:54.287 ServerApp] 500 POST /api/contents/work/Untitled.ipynb/checkpoints?1708592934278 (505591c04df545d2852106bc05e88733@192.168.65.1) 7.14ms referer=http://localhost:8888/lab/workspaces/auto-H/tree/work/Untitled.ipynb
[I 2024-02-22 09:09:43.054 ServerApp] Starting buffering for d8aa7554-fbf6-4a1d-b77e-4b162e539809:b5b0f8c7-f7c6-48cb-9e3c-143927a534b2
```

Reply:
Removing this part and changing to a named volume. Hopefully it's magically fixed without having to dig into what's causing it.